src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -219,6 +217,47 @@ func (client *Client) proxiedDialer(orig func(network, addr string) (net.Conn, e
 	}
 }
 
+func (client *Client) dialCONNECT(addr string, port int) (net.Conn, error) {
+	// Establish outbound connection
+	if client.shouldSendToProxy(port) {
+		log.Tracef("Proxying CONNECT request for %v", addr)
+		d := client.proxiedDialer(func(network, addr string) (net.Conn, error) {
+			// UGLY HACK ALERT! In this case, we know we need to send a CONNECT request
+			// to the chained server. We need to send that request from chained/dialer.go
+			// though because only it knows about the authentication token to use.
+			// We signal it to send the CONNECT here using the network transport argument
+			// that is effectively always "tcp" in the end, but we look for this
+			// special "transport" in the dialer and send a CONNECT request in that
+			// case.
+			return client.getBalancer().Dial("connect", addr)
+		})
+		return d("tcp", addr)
+	}
+	log.Tracef("Port not allowed, bypassing proxy and sending CONNECT request directly to %v", addr)
+	return protected.Dial("tcp", addr, 1*time.Minute)
```

Comment:
It's a bit unusual to have common code depending on a package used only on mobile. I don't have better option, though.